runtime/src/iree/builtins/ukernel/arch/x86_64/mmt4d_x86_64_avx2_fma.c

```diff
@@ -200,6 +200,81 @@ IREE_UK_MMT4D_TILE_FUNC_IMPL_FOR_M0(
     iree_uk_mmt4d_tile_s8s8s32_1x8x2_to_4x8x2_x86_64_avx2_fma,
     iree_uk_mmt4d_tile_s8s8s32_4x8x2_x86_64_avx2_fma, 4)
 
+IREE_UK_ATTRIBUTE_ALWAYS_INLINE static inline void
+iree_uk_mmt4d_tile_s8s4s32_1x8x4_to_8x8x4_x86_64_avx2_fma(
+    void* IREE_UK_RESTRICT out_tile, const void* IREE_UK_RESTRICT lhs_panel,
+    const void* IREE_UK_RESTRICT rhs_panel,
+    const iree_uk_mmt4d_params_t* params, int M0) {
+  IREE_UK_ASSERT(M0 >= 1 && M0 <= 8 && iree_uk_is_po2_u32(M0));
+  iree_uk_int32_t* IREE_UK_RESTRICT out_ptr = out_tile;
+  const iree_uk_int8_t* IREE_UK_RESTRICT lhs_ptr = lhs_panel;
+  const iree_uk_int8_t* IREE_UK_RESTRICT rhs_ptr = rhs_panel;
+  __m256i acc[8];
+  if (params->flags & IREE_UK_FLAG_MMT4D_ACCUMULATE) {
+    IREE_UK_UNROLL for (int i = 0; i < M0; ++i) {
+      acc[i] = _mm256_loadu_si256((__m256i*)(out_ptr + i * 8));
+    }
+  } else {
+    IREE_UK_UNROLL for (int i = 0; i < M0; ++i) {
+      acc[i] = _mm256_setzero_si256();
+    }
+  }
+
+  // Mask for extracting high nibbles.
+  __m128i low_nibble_mask = _mm_set1_epi16(0xf0f0);
```

Comment:
The naming here is also confusing in terms of "low" vs "high". `low_nibble_mask` is a mask to *mask out* the low bits, retaining the high bits. In other existing kernels, we name such masks "high masks", naming masks after what they retain, not what they mask out.

And example of a line of code that is made confusing by this terminology is at line 232 below,

```c
__m128i rhs_high_i8 = _mm_and_si128(rhs_i4, low_nibble_mask);
```

Actually I have another terminology topic. Why "low" and "high" are to the point here for the business of masking bits, past that point what really is to the point of what we are doing is "even" and "odd" lanes. It is only an implementation detail that the even (respectively odd) lanes are obtained by masking retaining only the low (respectively high) 4 bits in each byte.  Keeping low/high terminology throughout is confusing because that would usually indicate a splitting of the tile into two halves, not the lane-wise de-interleaving of odd/even lanes that we are doing here.